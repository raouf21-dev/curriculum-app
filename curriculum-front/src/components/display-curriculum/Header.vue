<template>
  <v-row>
    <v-col>
      <h1
        class="hover-icon-container"
        v-if="editField !== 'name'"
      >
        {{ selectedCurriculum.name }}
        <v-icon
          color="gray lighten-1"
          class="togglable-icon"
          @click="toggleEdit('name')"
        >
          mdi-pencil-box-outline
        </v-icon>
      </h1>
      <v-text-field
        v-else
        v-model="selectedCurriculum.name"
      >
        <template v-slot:append-outer>
          <v-btn
            small
            outlined
            color="primary"
            class="mr-1"
            @click="saveEdit('name')"
          >
            Save
          </v-btn>
          <v-btn outlined small @click="cancelEdit">
            Cancel
          </v-btn>
        </template>
      </v-text-field>

      <p
        class="hover-icon-container"
        v-if="editField !== 'goal'"
      >
        Goal: {{ selectedCurriculum.goal }}
        <v-icon
          color="gray lighten-1"
          class="togglable-icon"
          @click="toggleEdit('goal')"
          small
        >
          mdi-pencil-box-outline
        </v-icon>
      </p>
      <v-text-field
        v-else
        v-model="selectedCurriculum.goal"
      >
        <template v-slot:append-outer>
          <v-btn
            small
            outlined
            color="primary"
            class="mr-1"
            @click="saveEdit('goal')"
          >
            Save
          </v-btn>
          <v-btn outlined small @click="cancelEdit">
            Cancel
          </v-btn>
        </template>
      </v-text-field>

      <p
        class="hover-icon-container"
        v-if="editField !== 'description'"
      >
        {{ selectedCurriculum.description }}
        <v-icon
          color="gray lighten-1"
          class="togglable-icon"
          @click="toggleEdit('description')"
          small
        >
          mdi-pencil-box-outline
        </v-icon>
      </p>
      <v-text-field
        v-else
        v-model="selectedCurriculum.description"
      >
        <template v-slot:append-outer>
          <v-btn
            small
            outlined
            color="primary"
            class="mr-1"
            @click="saveEdit('description')"
          >
            Save
          </v-btn>
          <v-btn outlined small @click="cancelEdit">
            Cancel
          </v-btn>
        </template>
      </v-text-field>
    </v-col>
  </v-row>
</template>

<script>
export default {
  props: {
    editField: String,
    selectedCurriculum: Object,
    toggleEdit: Function,
    saveEdit: Function,
    cancelEdit: Function
  }  
}
</script>
