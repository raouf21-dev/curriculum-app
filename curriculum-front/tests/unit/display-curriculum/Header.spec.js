import { shallowMount } from '@vue/test-utils'
import Header from '@/components/display-curriculum/Header.vue'

describe('Header.vue', () => {
  test('if snapshot is correct', () => {
    const wrapper = shallowMount(Header, {
      propsData: {
        editField: '', // name, goal, or description
        selectedCurriculum: { name: 'Learn Vue.js' },
        toggleEdit: () => {},
        saveEdit: () => {},
        cancelEdit: () => {}
      }
    })
    expect(wrapper).toMatchSnapshot()
  })

  test('that title edit mode can display properly', () => {
    const wrapper = shallowMount(Header, {
      propsData: {
        editField: 'name',
        selectedCurriculum: { name: 'Learn Vue.js' },
        toggleEdit: () => {},
        saveEdit: () => {},
        cancelEdit: () => {}
      }
    })

    // TODO: find by a test id instead of class here
    // const input = wrapper.find('.name-edit-field')
    const input = wrapper.find('[data-test="name-edit-field"]')
    expect(input.exists()).toBe(true)
  })
})
