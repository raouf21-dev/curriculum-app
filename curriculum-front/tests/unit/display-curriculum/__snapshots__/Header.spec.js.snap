// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.vue if snapshot is correct 1`] = `
<v-row-stub tag="div">
  <v-col-stub tag="div">
    <h1 class="hover-icon-container">
      Learn Vue.js
      <v-icon-stub color="gray lighten-1" class="togglable-icon">
        mdi-pencil-box-outline
      </v-icon-stub>
    </h1>
    <div>
      <v-btn-stub color="info" tag="button" activeclass="" small="true" text="true" type="button">Add Goal</v-btn-stub>
    </div>
    <p class="hover-icon-container">

      <v-icon-stub color="gray lighten-1" small="" class="togglable-icon">
        mdi-pencil-box-outline
      </v-icon-stub>
    </p>
  </v-col-stub>
</v-row-stub>
`;
